columbo/_interaction.py

```diff
@@ -367,15 +380,31 @@ def __init__(
     def default(self) -> StaticOrDynamicValue[str]:
         return self._default
 
-    def validate(self, value: str, answers: Answers) -> Optional[str]:
+    def validate(self, value: str, answers: Answers) -> ValidationResponse:
         if self._validator is None:
             return None
         if callable(self._validator):
-            error = self._validator(value, answers)
-            if error:
-                return error
-            return None
-        raise ValueError(f"Invalid value for should_ask: {self._validator}")
+            result = self._validator(value, answers)
+            if isinstance(result, (ValidationSuccess, ValidationFailure)):
+                return result
+            else:  # handle deprecated, legacy validator responses
+                if result:
+                    return result
+                return None
```

Comment:
Could this be simplified to:
```suggestion
                return result
```
?

Thought being that the signature of the legacy validator being `Optional[str]`, `None` will implicitly be returned if not `result`.

Reply:
Good catch. This could be simplified, but I've updated the validator function so this is no longer a problem.